packages/embark/src/cmd/cmd_controller.js

```diff
@@ -408,13 +409,30 @@ class EmbarkController {
 
   }
 
-  async reset() {
-    const doneMessage = __("reset done!").green;
-    await embarkReset({doneMessage});
+  async reset(options) {
+    const embarkConfig = require(fs.dappPath(options.embarkConfig || 'embark.json'));
+
+    let removePaths = [];
+    let defaultPaths = [...defaultResetPaths];
+
+    defaultPaths.push(embarkConfig.buildDir);
+    embarkConfig.generationDir && defaultPaths.push(embarkConfig.generationDir);
+
+    if (embarkConfig.options && embarkConfig.options.reset) {
+      if (embarkConfig.options.reset.defaults) {
+        removePaths = removePaths.concat(defaultPaths);
+      }
+      if (embarkConfig.options.reset.files) {
+        removePaths = removePaths.concat(embarkConfig.options.reset.files);
+      }
+    } else {
+      removePaths = defaultPaths;
+    }
+    removePaths = [...new Set(removePaths.map(path => path.charAt(path.length - 1) === '/' ? path.substr(0, path.length - 1) : path))];
```

Comment:
This might look a little bananas here, but basically what we're doing is:

- trimming of trailing `/` from paths, this is because `buildDir: 'dist/'` while embark-reset's default paths include `dist`. Adding `buildDir` to defaults would result in `['dist', 'dist/']`. We trim this out so we can...
- Create a new array from `Set()` ensure duplicates are removed

Reply:
Good approach with `Set`, I used the same. `rimraf` doesn't care about trailing slashes so not strictly necessary to trim them.

Reply:
Yes, rimraf doesn't care, however since https://github.com/embark-framework/embark/commit/3988fb4c8ac18c7605968c9dca373ed126522563 we output all the paths that are going to be removed. Users might be confused that it has `dist` and `dist/` in there.